package NYT.NRpcChunkHolder;

////////////////////////////////////////////////////////////////////////////////

message TReqStartChunk
{
    required bytes ChunkId = 1;
}

message TRspStartChunk
{
}

////////////////////////////////////////////////////////////////////////////////

message TReqFinishChunk
{
    required bytes ChunkId = 1;
}

message TRspFinishChunk
{
}

////////////////////////////////////////////////////////////////////////////////

message TReqPutBlocks
{
    required bytes ChunkId = 1;
    required int32 StartBlockIndex = 2;
    required int64 StartOffset = 3;
}

message TRspPutBlocks
{
}

////////////////////////////////////////////////////////////////////////////////

message TReqSendBlocks
{
    required bytes ChunkId = 1;
    required string Destination = 2;
    required int32 StartBlockIndex = 3;
    required int32 EndBlockIndex = 4;

}

message TRspSendBlocks
{
}

////////////////////////////////////////////////////////////////////////////////

message TReqFlushBlocks
{
    required bytes ChunkId = 1;
    required int32 StartBlockIndex = 2;
    required int32 EndBlockIndex = 3;
}

message TRspFlushBlocks
{
}

////////////////////////////////////////////////////////////////////////////////

message TBlockInfo
{
    required int32 Offset = 1;
    required int32 Size = 2;
}

message TReqGetBlocks
{
    required bytes ChunkId = 1;
    repeated TBlockInfo Blocks = 2;
    optional bool IsPrecharge = 3 [default = false];
}

message TRspGetBlocks
{
}

